lib/web/scripts/node-join/install.sh

```diff
@@ -480,6 +481,39 @@ db_service:
 {{end}}
 EOF
 }
+
+# installs the provided teleport config (for Discovery Service)
+install_teleport_discovery_config() {
+    log "Writing Teleport discovery service config to ${TELEPORT_CONFIG_PATH}"
+    CA_PINS_CONFIG=$(get_yaml_list "ca_pin" "${CA_PIN_HASHES}" "  ")
+
+    # This file is processed by `shellschek` as part of the lint step
+    # It detects an issue because of un-set variables - $index and $line. This check is called SC2154.
+    # However, that's not an issue, because those variables are replaced when we run go's text/template engine over it.
+    # When executing the script, those are no long variables but actual values.
+    # shellcheck disable=SC2154
+    cat << EOF > ${TELEPORT_CONFIG_PATH}
+version: v3
+teleport:
+  nodename: ${NODENAME}
+  auth_token: ${JOIN_TOKEN}
+${CA_PINS_CONFIG}
+  proxy_server: ${TARGET_HOSTNAME}:${TARGET_PORT}
+  log:
+    output: stderr
+    severity: INFO
+auth_service:
+  enabled: no
+ssh_service:
+  enabled: no
+proxy_service:
+  enabled: no
+discovery_service:
+  enabled: "yes"
+  discovery_group: "{{.discoveryGroup}}"
```

Comment:
without any matchers (e.g. `aws`, `azure`), this discovery service doesn't do anything. Is this intended?

Reply:
It should be capable of watch DiscoveryConfig resource change events and, if the `discovery_group` matches, dynamically load them.

Reply:
👍 Nice, I didn't know it can do that.

Reply:
It's quite recent
Related issue:
https://github.com/gravitational/teleport/issues/25494